geocity/apps/submissions/payments/postfinance/processor.py

```diff
@@ -0,0 +1,172 @@
+from django.conf import settings
+from django.urls import reverse
+from django.utils import timezone
+from postfinancecheckout import (
+    Configuration,
+    Environment,
+    LineItemAttribute,
+    TransactionEnvironmentSelectionStrategy,
+    TransactionState,
+)
+from postfinancecheckout.api import (
+    TransactionPaymentPageServiceApi,
+    TransactionServiceApi,
+)
+from postfinancecheckout.models import LineItem, LineItemType, TransactionCreate
+
+from ..processors import PaymentProcessor
+from .models import PostFinanceTransaction
+
+
+class PostFinanceCheckoutProcessor(PaymentProcessor):
+    name = "PostFinance Checkout"
+    identifier = "PostFinance"
+    transaction_class = PostFinanceTransaction
+    required_settings = [
+        "PAYMENT_PROCESSING_TEST_ENVIRONMENT",
+    ]
+    required_params = ["space_id", "user_id", "api_key"]
+
+    def __init__(self, space_id, user_id, api_key):
+        super().__init__()
+        self.config = Configuration(
+            user_id=user_id,
+            api_secret=api_key,
+        )
+        self.space_id = space_id
+        self.transaction_service_api = None
+        self.transaction_payment_page_service = None
+
+    def _get_transaction_service_api(self):
+        if getattr(self, "transaction_service_api", None) is None:
+            self.transaction_service_api = TransactionServiceApi(
+                configuration=self.config
+            )
+        return self.transaction_service_api
+
+    def _get_transaction_payment_page_service_api(self):
+        if getattr(self, "transaction_payment_page_service", None) is None:
+            self.transaction_payment_page_service = TransactionPaymentPageServiceApi(
+                configuration=self.config
+            )
+        return self.transaction_payment_page_service
+
```

Comment:
@danickfort il faudrait que tu m'expliques pourquoi on utilise la "transaction service api" et la "payment page api". Il me semblait que cétait l'une ou l'autre :)

Reply:
This is how the PostFinance API works, there is an "API" for creating a transaction on their side, and then a "payment page API" to get the relevant payment page URL. It is directly taken from their example on the README page, so I trust that it is the right way to go 😄